File: core/src/main/kotlin/com/bing/lan/core/base/query/IpLogQueryObject.java
package com.bing.lan.core.base.query;

import com.bing.lan.core.base.utils.DateUtil;

import org.springframework.format.annotation.DateTimeFormat;

import java.util.Date;

import lombok.Getter;
import lombok.Setter;

/**
 * IpLog查询对象
 */
@Setter
@Getter
public class IpLogQueryObject extends QueryObject {

    private Date beginDate;
    private Date endDate;
    private String username;
    private int userType = -1;
    private boolean like;
    private int state = -1;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    public void setBeginDate(Date beginDate) {
        this.beginDate = beginDate;
    }

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }

    public Date getEndDate() {
        if (endDate != null) {
            return DateUtil.endOfDay(endDate);
        }
        return null;
    }
}

File: core/src/main/kotlin/com/bing/lan/core/base/query/IpLogQueryObject.kt
package com.bing.lan.core.base.query

import com.bing.lan.core.base.utils.DateUtil
import lombok.Getter
import lombok.Setter
import org.springframework.format.annotation.DateTimeFormat
import java.util.*

/**
 * Created by 蓝兵 on 2018/6/1.
 */
/**
 * IpLog查询对象
 */
@Setter
@Getter
open class IpLogQueryObject : QueryObject() {

    @set:DateTimeFormat(pattern = "yyyy-MM-dd")
    var beginDate: Date? = null
    @set:DateTimeFormat(pattern = "yyyy-MM-dd")
    var endDate: Date? = null
        get() = if (field != null) {
            DateUtil.endOfDay(field)
        } else null
    var username: String? = null
        set(username) {
            field = this.username
        }
    var userType = -1
        set(userType) {
            field = this.userType
        }
    var isLike: Boolean = false
        set(like) {
            field = isLike
        }
    var state = -1
        set(state) {
            field = this.state
        }
}
